### PR TITLE
fix: CVE-2022-1996 & CVE-2022-29526

### PR DESCRIPTION
**Pull request by umagnus**
fix CVE-2022-1996 & CVE-2022-29526

**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

fix: CVE-2022-1996 & CVE-2022-29526

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

```
local-provisioner (gobinary)
============================
Total: 2 (UNKNOWN: 0, LOW: 0, MEDIUM: 1, HIGH: 0, CRITICAL: 1)

┌────────────────────────────────┬────────────────┬──────────┬────────────────────────────────────┬───────────────────────────────────┬──────────────────────────────────────────────────────────────┐
│            Library             │ Vulnerability  │ Severity │         Installed Version          │           Fixed Version           │                            Title                             │
├────────────────────────────────┼────────────────┼──────────┼────────────────────────────────────┼───────────────────────────────────┼──────────────────────────────────────────────────────────────┤
│ github.com/emicklei/go-restful │ CVE-2022-1996  │ CRITICAL │ v2.9.5+incompatible                │ v3.8.0                            │ go-restful: Authorization Bypass Through User-Controlled Key │
│                                │                │          │                                    │                                   │ https://avd.aquasec.com/nvd/cve-2022-1996                    │
├────────────────────────────────┼────────────────┼──────────┼────────────────────────────────────┼───────────────────────────────────┼──────────────────────────────────────────────────────────────┤
│ golang.org/x/sys               │ CVE-2022-29526 │ MEDIUM   │ v0.0.0-20220209214540              │ 0.0.0-202204122112                │ golang: syscall: faccessat checks wrong group                │ 
│                                │                │          │                                    │                                   │ https://avd.aquasec.com/nvd/cve-2022-29526                   │
└────────────────────────────────┴────────────────┴──────────┴────────────────────────────────────┴───────────────────────────────────┴──────────────────────────────────────────────────────────────┘
```

fix CVE-2022-1996: update github.com/emicklei/go-restful to v3.8 or higher, add `replace github.com/emicklei/go-restful => github.com/emicklei/go-restful/v3 v3.8.0 ` in `go.mod`
fix CVE-2022-29526: update `golang.org/x/sys v0.0.0-20220412211240-33da011f77ad` in `go.mod`

**Release note**:
```release-note
fix: CVE-2022-1996
fix: CVE-2022-29526
```

### CODE REVIEW

**Comment by umagnus** [comment]
Hello, I have fixed CVE-2022-1996 & CVE-2022-29526, could you approve it, thanks. @mauriciopoppe

File: pkg/deleter/deleter.go [hunk 1]
```diff
@@ -336,15 +336,15 @@ func (d *Deleter) execScript(pvName string, blkdevPath string, exe string, exeAr
 
 // runJob runs a cleaning job.
 // The advantages of using a Job to do block cleaning (which is a process that can take several hours) is as follows
-// 1) By naming the job based on the specific name of the volume, one ensures that only one instance of a cleaning
```
Comment (mauriciopoppe) [inline pkg/deleter/deleter.go:339]: This change doesn't look like it's part of the CVE changes

File: go.mod [hunk 2]
```diff
@@ -105,6 +106,7 @@ require (
 )
 
 replace (
+	github.com/emicklei/go-restful => github.com/emicklei/go-restful/v3 v3.8.0
```
Comment (mauriciopoppe) [inline go.mod:109]: what's the reason behind pinning this package?

**Review by mauriciopoppe — COMMENTED** [review]
I think all the commits should be squashed into 1

/assign @msau42

File: pkg/deleter/deleter.go (hunk 1 above)
Comment (umagnus) [reply pkg/deleter/deleter.go:339]: It is to fix the gofmt issue.

File: go.mod (hunk 2 above)
Comment (umagnus) [reply go.mod:109]: To fix the CVE-2022-1996, we must use go-restful v3.8 or higher, but the metrics.go 0.24 import github.com/emicklei/go-restful, so we replace it to v3

**Comment by umagnus** [comment]
> I think all the commits should be squashed into 1
> 
> /assign @msau42

**Comment by umagnus** [comment]
Hello, I have squashed all the commits into 1. @mauriciopoppe

**Comment by mauriciopoppe** [comment]
Unfortunately I don't have /approve permissions.

/assign @andyzhangx

**Review by andyzhangx — APPROVED** [review]
/lgtm
/approve

**Pull request merged**